### PR TITLE
CMasternode::GetLastPaid cleaning another strange piece of "code"

### PR DESCRIPTION
**Pull request by furszy**
`CMasternode::GetLastPaid`, instead of using the chain tip index, and loop backwards to **only** obtain the block height, use the cached height.

### CODE REVIEW

File: src/masternode.h [hunk 1]
```diff
@@ -210,7 +210,7 @@ class CMasternode : public CSignedMessage
         READWRITE(nLastScanningErrorBlockHeight);
     }
 
-    int64_t SecondsSincePayment();
+    int64_t SecondsSincePayment(int nChainHeight);
```
Comment (random-zebra) [inline src/masternode.h:213]:
```suggestion
    int64_t SecondsSincePayment(int nChainHeight) const;
```

File: src/masternode.h [hunk 2]
```diff
@@ -260,7 +260,7 @@ class CMasternode : public CSignedMessage
         return strStatus;
     }
 
-    int64_t GetLastPaid();
+    int64_t GetLastPaid(int nChainHeight);
```
Comment (random-zebra) [inline src/masternode.h:263]:
```suggestion
    int64_t GetLastPaid(int nChainHeight) const;
```

**Review by random-zebra — CHANGES_REQUESTED** [review]
This doesn't  look right.
The `nBlockHeight`argument of `CMasternodeMan::GetNextMasternodeInQueueForPayment`, that here is passed to `CMasternode::SecondsSincePayment`, is not necessarily the chain height. 
In fact, most of the times it isn't, it could even be **higher** (as we use it to get the hash at `height - 101`)... e.g. see the call to `masternodePayments.ProcessBlock` in validation.cpp).

In `CMasternode::GetLastPaid()`, instead, we want to start exactly at the tip, and go backwards, to find the block where a particular masternode was paid.
I think that a better approach would be to move this function to the manager, returning the height only (e.g. `int CMasternodeMan::GetBlockLastPaid(const CTxIn& mnode_vin)`).
Then access the chainActive, and get the time, outside of it, without any other lock held.
Of course this would require to implement something like `int64_t CMasternode::GetOffset()` too.

**Comment by furszy** [comment]
Good point, will rework it differently.

**Pull request closed without merging**